[ui][ios] Add Background component (#48904)
expo/expo: docs/pages/versions/unversioned/sdk/ui/swift-ui/background.mdx

---
title: Background
description: A SwiftUI Background component for layering content behind another view.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-ui'
packageName: '@expo/ui'

> platforms: ['ios', 'tvOS', 'expo-go']

Reviewer: 💡 **nit** `mdx-components-sdk.md#platformtags`

The canonical value is lowercase `tvos`: 35 pages in this directory use it, and only **overlay.mdx** carries the mixed-case copy this was based on.

```suggestion
platforms: ['ios', 'tvos', 'expo-go']
```

<!-- docs-review -->

---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';

> Expo UI Background matches the official SwiftUI [background](<https://developer.apple.com/documentation/swiftui/view/background(alignment:content:)>) modifier and draws another view behind a view, positioned with a specified alignment. The base view keeps its own size, and the background content is laid out behind it.

Reviewer: ✏️ **style** `review-ai-text.md#t5-restatement`

"the background content is laid out behind it" repeats what the first sentence already said, and "another view behind a view" is hard to follow. Keep the layout fact and name the slot content the way the rest of the page and the prop description do.

```suggestion
Expo UI Background matches the official SwiftUI [background](<https://developer.apple.com/documentation/swiftui/view/background(alignment:content:)>) modifier and draws background content behind a view, positioned with a specified alignment. The base view keeps its own size.
```

<!-- docs-review -->

> > **info** To fill a view with a solid color instead of a view, use the `background` modifier from `@expo/ui/swift-ui/modifiers`.

Reviewer: ✏️ **style** `links-and-headings.md#internal-links`

Link the modifier you are pointing at, the way **image.mdx** and **list.mdx** do in this directory. Drop "instead of a view" as well, which reads as though the color replaces a view.

```suggestion
> **info** To fill a view with a solid color, use the [`background`](modifiers/#backgroundcolor-shape) modifier from `@expo/ui/swift-ui/modifiers`.
```

<!-- docs-review -->

## Installation

<APIInstallSection />

## Usage

### Basic background

```tsx BackgroundExample.tsx
import { Background, Host, RoundedRectangle, Text } from '@expo/ui/swift-ui';
import { font, foregroundStyle, padding } from '@expo/ui/swift-ui/modifiers';

export default function BackgroundExample() {
  return (
    <Host style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Background>
        <Text modifiers={[font({ size: 17, weight: 'semibold' }), padding({ all: 16 })]}>
          Card title
        </Text>
        <Background.Content>
          <RoundedRectangle cornerRadius={12} modifiers={[foregroundStyle('#E8F0FE')]} />
        </Background.Content>
      </Background>
    </Host>
  );
}
```

> ### Aligning the background content

Reviewer: 💡 **nit** `links-and-headings.md#titles-and-headings`

Usage headings in this directory are short noun phrases, so this one reads long beside "Basic background".

```suggestion
### Background alignment
```

<!-- docs-review -->
